controllers/storagecluster/cephcluster.go

```diff
@@ -110,12 +110,7 @@ func (obj *ocsCephCluster) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.
 	var cephCluster *rookCephv1.CephCluster
 	// Define a new CephCluster object
 	if sc.Spec.ExternalStorage.Enable {
-		extRArr, err := r.retrieveExternalSecretData(sc)
-		if err != nil {
-			r.Log.Error(err, "Could not retrieve the External Cluster details.",
-				"CephCluster", klog.KRef(sc.Namespace, sc.Name))
-			return reconcile.Result{}, err
-		}
+		extRArr := externalOCSResources[sc.UID]
```

Comment:
Don't we need to check the err like before?

Reply:
no, we don't need to as we are setting the value in the map and will get errors while setting the value itself and returning it.

Reply:
OK, but don't we need to check externalOCSResources[sc.UID] is not nil?

Reply:
we can do that, but it should not be nil

Reply:
done